From e13f235ea5e643b900cc3017c108167a072d5af7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:41:09 +0800
Subject: [PATCH] Revert "serial_core: Add wake_peer uart operation which is
 called before starting UART TX. The idea here is to provide a mechanism where
 we can wakeup our peer before sending data."

This reverts commit f5353976c6a13732718b472ad308af1bb997f025.
---
 drivers/serial/serial_core.c | 3 ---
 include/linux/serial_core.h  | 1 -
 2 files changed, 4 deletions(-)

diff --git a/drivers/serial/serial_core.c b/drivers/serial/serial_core.c
index 47ad0cb20e42..dcc72444e8e7 100644
--- a/drivers/serial/serial_core.c
+++ b/drivers/serial/serial_core.c
@@ -94,9 +94,6 @@ static void __uart_start(struct tty_struct *tty)
 	struct uart_state *state = tty->driver_data;
 	struct uart_port *port = state->uart_port;
 
-	if (port->ops->wake_peer)
-		port->ops->wake_peer(port);
-
 	if (!uart_circ_empty(&state->xmit) && state->xmit.buf &&
 	    !tty->stopped && !tty->hw_stopped)
 		port->ops->start_tx(port);
diff --git a/include/linux/serial_core.h b/include/linux/serial_core.h
index 8bdb2548ae3d..e83a0ad06d32 100644
--- a/include/linux/serial_core.h
+++ b/include/linux/serial_core.h
@@ -220,7 +220,6 @@ struct uart_ops {
 	void		(*pm)(struct uart_port *, unsigned int state,
 			      unsigned int oldstate);
 	int		(*set_wake)(struct uart_port *, unsigned int state);
-	void		(*wake_peer)(struct uart_port *);
 
 	/*
 	 * Return a string describing the type of the port
-- 
2.35.3

